html/js/product-preferences.js

```diff
@@ -23,6 +23,32 @@ function get_user_product_preferences() {
 	return user_product_preferences;
 }
 
+function getFlavorFromHostname() { // get the flavor by the url 
+    var host = window.location.hostname;
+
+    if (host.includes("openbeautyfacts")) {
+        return "obf";
+    }
+    if (host.includes("openfoodfacts")) {
+        return "off";
+    }
+    
+    return "off"; // default 
+}
+
+var flavor = getFlavorFromHostname();
+
+var reset_message;
+
+if (flavor === "off") { // get the reset message depend on the flavor
```

Comment:
Instead of having logic like this, I think it would be better to have logic like:
if we have a string with the id reset_preferences_details + product_type (it's better to use the "food" and "beauty" product type instead of the flavors "obf" and "off" as at some point we might merge websites) -> use it
otherwise, if there is no string defined (and not empty), use the default message.